### DISCUSSION TITLE
Background image cannot be changed

### DISCUSSION THREAD

**Discussion by johannes1984** [linuxserver/Heimdall#996] [Q&A] [answered]
Using version 2.4.11 I cannot change the background image. I open the dialogue to chose the new file, select a jpg and then hit the "Save" button. Then it brings me back to the settings overview, confirms "You have successfully edited this setting" but still states "- not set -". And the background remains the standard image.

**Comment by mariusberget92** [comment]
Same.

**Comment by thedinz** [comment]
I hvae had this issue for so long that I just upload it via FTP manually and rename it to the default image name, works for me, I don't change it often enough for that to be too cumbersome.

**Comment by deviantintegral** [comment, marked as answer]
Did you see the notes in the README at https://github.com/linuxserver/Heimdall#installing or https://github.com/linuxserver/Heimdall#new-background-image-not-being-set? For me bumping the max upload size fixed this.

What's unclear to me is why this isn't a default setting in the docker containers. I would have expected an 8 or 16MB limit would cover most images.

**Comment by aptalca** [comment]
> What's unclear to me is why this isn't a default setting in the docker containers. I would have expected an 8 or 16MB limit would cover most images.

Because it's the php default, same as a bare metal install. And we make it user customizable in the container

**Comment by deviantintegral** [comment]
Thanks. I filed https://github.com/linuxserver/docker-heimdall/issues/100 because we don't know if the original reporter is using Docker or not.

**Comment by johannes1984** [comment]
> Thanks. I filed [linuxserver/docker-heimdall#100](https://github.com/linuxserver/docker-heimdall/issues/100) because we don't know if the original reporter is using Docker or not.

Just to confirm that I use the Docker version

**Comment by HannesJo0139** [comment]
@aptalca I just encountered that changing `upload_max_filesize` only is not working anymore. Now I also have to set `post_max_size`. Maybe you could change the docs according?

**Comment by 9021007** [comment]
I am still having this issue, despite setting my `upload_max_filesize` and `post_max_size` to 30M.

**Comment by HannesJo0139** [comment]
> I am still having this issue, despite setting my `upload_max_filesize` and `post_max_size` to 30M.

Depending on your connection speed, `max_input_time` and `max_execution_time` may also have to be raised.

**Comment by arjunsrinivasan1997** [comment]
I have tried setting `upload_max_filesize` `post_max_size`, `max_input_time`, and `max_execution_time`, and am unable to set a new background image

**Comment by rubenixnagios** [comment]
same on linuxserver/heimdall version 2.4.13, will try to rollback to a previous version just to check...
anyone found a fix for that?

**Comment by HannesJo0139** [comment]
I cannot reproduce. Are you using a proxy? It would need the same settings I think.

**Comment by rubenixnagios** [comment]
> same on linuxserver/heimdall version 2.4.13, will try to rollback to a previous version just to check... anyone found a fix for that?

my bad sorry; just a permissions issue! background changed succesfully

**Comment by arjunsrinivasan1997** [comment]
> I cannot reproduce. Are you using a proxy? It would need the same settings I think.


Taking it off the proxy and tweaking `upload_max_filesize `and  `post_max_size` fixed the issue for me.

**Comment by B1GWH1T3** [comment]
Running - Docker > [lscr.io/linuxserver/heimdall:latest]
Set the config file php-local.ini with upload_max_filesize = 30M > did not work since I got an error 401
searched more and found this setting post_max_size as indicated by other users, set that and working with no issues now

If you using the linuxserver git the file to modify will be in 

    volumes:
      - /path/to/appdata/config:/config
      
       containername/php/php-local.ini
       
 
#php-local.ini content
 ; Edit this file to override php.ini directives and restart the container

date.timezone = Africa/Johannesburg
upload_max_filesize = 30M
post_max_size = 30M

**Comment by lazarus477** [comment]
I tried every solution provided here and I am still unable to change the background image, short of swapping out the app default background image file with another one.

**I have max file and post sizes set to 100M in php.ini and max execution time set to 360.**

**System**:

- OS: ArchLinux
- Heimdall v 2.5.7
- Server Version: Apache/2.4.57 (Unix) PHP/8.2.5
- Server MPM: prefork
- Server Built: Apr 17 2023 14:10:16

**Reply by B1GWH1T3** [reply]
Did you use the correct config file?

**Reply by lazarus477** [reply]
@B1GWH1T3 I cannot recall having edited/modified any **config** file during the setup of Heimdall. When going back through the documentation at: https://github.com/linuxserver/Heimdall, I noticed this;
### Installing
...
"_If you find you can't change the background make sure php_fileinfo is enabled in your php.ini. I believe it should be by default, but one user came across the issue on a windows system._"
...

Now I am  looking into the possibility that I am missing a **PHP module**.

**Reply by lazarus477** [reply]
From what I can tell the P**HP module fileinfo** is not available in the regular arch repositories but provided through **Aur**. So I do not have **fileinfo** installed and configured for use. This could explain my issue.

**Reply by lazarus477** [reply]
**Current experience**:
I can choose to change the background in the dashboard setting. Doing so opens up a file manager, I select the image file (jpg) and it 's thumbnail is shown. I press save. When presented with the dashboard all I see is a plain background color. Going back to the settings I can still see the thumbnail of my uploaded image, it has just not replaced the default.
